stash_engine/app/models/stash_engine/concerns/string_enum.rb

```diff
@@ -0,0 +1,65 @@
+require 'active_support/concern'
+
+# Provides support for enums stored as Strings in the DB
+# This can be dropped in Rails 5
+module StashEngine
+
+  module Concerns
+
+    module StringEnum
+
+      extend ActiveSupport::Concern
+
+      class_methods do
+
+        # Disabling rubocop here because we reasonably abstract this into callable methods
+        # due to the
```

Comment:
due to the ???

Reply:
Ha ha. Guess I must have gone off on my lunch break there ;)

I broke the one large function into smaller chunks to appease the Rubocop gods. Will remove that comment